isis/src/tgo/apps/tgocassisrdrgen/tgocassisrdrgen.cpp

```diff
@@ -57,22 +57,22 @@ void IsisMain() {
   // by the user, set it to the Observation Id.
   // This is added before the translation instead of adding it to the exported xml
   // because of the ease of editing pvl vs xml.
-  PvlGroup &instrumentGroup = label->findObject("IsisCube").findGroup("Instrument");
+  PvlGroup &targetGroup = isMosaic ? label->findObject("IsisCube").findGroup("Mosaic")
```

Comment:
Why did you chose to do this this way?

Reply:
I originally had a method called writeProductId that would take in a group name, either Instrument or Mosaic. I would either have to pass in the user interface and the input cube label as parameters or have those as global variables. I was not sure about how this was usually done, so I did it this way instead. We also can not do an if-else where we check to see if isMosaic is true and then set targetGroup to find the Mosaic and else find the Instrument group because we can not have an uninitialized reference.